module-utils/tar/tar.cpp

```diff
@@ -14,23 +13,25 @@ namespace
         return entry.name() != "." and entry.name() != ".." and entry.name() != "...";
     }
 
-    void write_to_file(const tar::entry &entry, const std::filesystem::path &path)
+    void writeToFile(const tar::entry &entry, const std::filesystem::path &path)
     {
-        constexpr std::size_t chunk_size = 1024 * 128;
-        std::ofstream out_file{path, std::ios::binary};
-        if (not out_file.is_open()) {
+        auto fd = std::fopen(path.c_str(), "wb");
```

Comment:
Why do you use `std::fopen` instead of `std::ofstream`? The `ofstream` uses RAII so you don't need to take care of freeing resources if something goes wrong.

Reply:
I was experimenting with adding `setvbuf` to this file stream too (but it didn't speed up the process at all) and just left it that way :wink: 
I can change it back to `std::ofstream` if you prefer, but IMO this change shouldn't create any issues in case of failure - if the file descriptor is successfully created, it will be closed at the end of this method, if opening fails, we just get a `nullptr` and throw an exception, there's nothing to be closed. Additionally, after the change all file accesses during update process are performed using the same API :wink:

Reply:
Ok, I get it. So leave `std::fopen` ;)